newrelic/data_source_newrelic_synthetics_private_location.go

```diff
@@ -77,5 +84,28 @@ func dataSourceNewRelicSyntheticsPrivateLocationRead(ctx context.Context, d *sch
 		return diag.FromErr(err)
 	}
 
+	key := fetchPrivateLocationKey(location.Tags)
+	if len(key) == 0 {
+		//logs the absence of a key but does not throw an error to prevent halting execution
+		log.Printf("[INFO] No keys found corresponding to the queried private location.")
+	}
```

Comment:
For the reviewer: this block has been added to cover a case in which the "key" tag is not found in the list of tags in the response of the NG query (just in case). The idea is to not halt execution (and stop rest of the attributes from being exported) if the key is absent for some reason.

Any thoughts are welcome 🙂